services/common/src/components/common/FileUpload.js

```diff
@@ -0,0 +1,280 @@
+/* eslint-disable */
+import React from "react";
+import PropTypes from "prop-types";
+import "filepond-polyfill";
+import { FilePond, registerPlugin } from "react-filepond";
+import { notification } from "antd";
+import { invert, uniq } from "lodash";
+import "filepond/dist/filepond.min.css";
+import FilePondPluginFileValidateSize from "filepond-plugin-file-validate-size";
+import FilePondPluginFileValidateType from "filepond-plugin-file-validate-type";
+import * as tus from "tus-js-client";
+import { ENVIRONMENT } from "@mds/common";
+import { APPLICATION_OCTET_STREAM } from "@mds/common/constants/fileTypes";
+import { createRequestHeader } from "@common/utils/RequestHeaders";
+import { bindActionCreators } from "redux";
+import { pollDocumentUploadStatus } from "@mds/common/redux/actionCreators/documentActionCreator";
+import { connect } from "react-redux";
+
+registerPlugin(FilePondPluginFileValidateSize, FilePondPluginFileValidateType);
+
+const propTypes = {
+  uploadUrl: PropTypes.string.isRequired,
+  maxFileSize: PropTypes.string,
+  acceptedFileTypesMap: PropTypes.objectOf(PropTypes.string),
+  onFileLoad: PropTypes.func,
+  onAbort: PropTypes.func,
+  onRemoveFile: PropTypes.func,
+  addFileStart: PropTypes.func,
+  chunkSize: PropTypes.number,
+  allowRevert: PropTypes.bool,
+  allowMultiple: PropTypes.bool,
+  allowReorder: PropTypes.bool,
+  maxFiles: PropTypes.number,
+  afterSuccess: PropTypes.shape({
+    action: PropTypes.func,
+    projectGuid: PropTypes.string,
+    irtGuid: PropTypes.string,
+  }),
+  labelIdle: PropTypes.string,
+  onprocessfiles: PropTypes.func,
+  importIsSuccessful: PropTypes.func,
+  beforeAddFile: PropTypes.func,
+  beforeDropFile: PropTypes.func,
+  itemInsertLocation: PropTypes.func | PropTypes.string,
+  notificationDisabledStatusCodes: PropTypes.arrayOf(PropTypes.number),
+  shouldReplaceFile: PropTypes.bool,
+  replaceFileUploadUrl: PropTypes.string,
+  file: PropTypes.object,
+  shouldAbortUpload: PropTypes.bool,
+  onAfterResponse: PropTypes.func,
+  pollDocumentUploadStatus: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  maxFileSize: "750MB",
+  acceptedFileTypesMap: {},
+  onFileLoad: () => {},
+  onAbort: () => {},
+  onRemoveFile: () => {},
+  addFileStart: () => {},
+  chunkSize: 1048576, // 1MB
+  allowRevert: false,
+  allowMultiple: true,
+  maxFiles: null,
+  afterSuccess: null,
+  onprocessfiles: () => {},
+  importIsSuccessful: () => {},
+  labelIdle: 'Drag & Drop your files or <span class="filepond--label-action">Browse</span>',
+  beforeAddFile: () => {},
+  beforeDropFile: () => {},
+  itemInsertLocation: "before",
+  notificationDisabledStatusCodes: [],
+  shouldReplaceFile: false,
+  replaceFileUploadUrl: "",
+  file: null,
+  shouldAbortUpload: false,
+  onAfterResponse: () => {},
+};
+
+class FileUpload extends React.Component {
```

Comment:
Interesting.  I really like the idea of having the fileUpload component moved to the common package, but I don't think they behave in exactly the same way.  

We don't need to remove this, but we'll need to consider rendering the differences seperately for each FE if we want to use this in Core.